tools/wsdl2aws-generator-stub.adb

```diff
@@ -488,6 +496,14 @@ package body Stub is
          Text_IO.New_Line (Stub_Adb);
       end if;
 
+      if O.Gen_CB then
+         Text_IO.Put_Line
+            (Stub_Adb,
+             "      Post_Call_Callback (Connection, " &
+                """" & To_String (O.Prefix) & Proc & """, " &
+                "Payload, Response, Schema);");
```

Comment:
note that Schema is a global variable already available to users. I would not pass it here.

Reply:
Can I just use SOAP.WSDL.Schema.Definition?
The callback might not know which request caused the callback.

Reply:
No, the Schema variable is generated at the top of the package hierarchy. See root package of a generated code. And, BTW, the Schema variable above passed to Post_Call_Callback is exactly this variable. Note also that the Schema variable contains the definitions for the whole Web Service and is never modified after the initialisation phase (elaboration).

Reply:
I see a Schema definition in the top of the package hierarchy of a specific service-set only. We have multiple service-sets aggregated with awsascb, but a common handler for callbacks across these service-sets.

Reply:
Indeed, I was thinking about this. So let's keep Schema in the callback.